docs/advanced/analytics/plugins.mdx

```diff
@@ -0,0 +1,340 @@
+      return !!window.myPluginLoaded;
+    },
+  };
+}
+```
+
+### Extending with `EcommercePlugin`
+
+The `EcommercePlugin` plugin will allow you to create an interface to map
+[e-commerce events and properties](https://developers.google.com/analytics/devguides/collection/ga4/ecommerce)
+to the relevant events and properties for your `acme` provider, you can find a
+full list of events in the
+[EcommercePlugin.js](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/src/web/core/analytics/plugins/e-commerce/domain/e-commerce-plugin.js)
+
+Below is an example of how to extend the `analytics-plugin-acme` plugin to
+handle e-commerce events.
+
+```js title="my-analytics-module/plugins/analytics-plugin-acme.js"
+import { EcommercePlugin } from "web/core/analytics/plugins/e-commerce";
+
+// Extend the EcommercePlugin with your custom overrides
+class AcmeEcommercePlugin extends EcommercePlugin {
+  // The event we use during tracking in the case is "Product Viewed"
+  productViewed = (payload) => {
+    return {
+      // Use the event name required by acme
+      event: "acme_view_item",
+      // Map the props to the required props of acme
+      properties: {
+        product_id: payload.sku,
+        product_name: payload.name,
+        product_currency: payload.currency,
+        product_price: payload.price,
+      },
+    };
+  };
+
+  // You can also add other events like "Product Reviewed", "Product Shared".
+  customEventsAndProperties = {
+    // the event key used in your `trackEvent` call
+    "Product Reviewed": (properties) => {
+      return {
+        // the tracking method will be called with this event
+        event:"acme_product_reviewed",
+        // the tracking method will be called with these properties
+        properties: {
+           product_id: properties.sku,
+           review_rating: properties.rating,
+        },
+      }
+    },
+    "Product Shared": (properties) => {
+        return {
+          event:"acme_product_shared",
+          properties: {
+             product_id: properties.sku,
+             method: properties.method,
+          },
+        }
+    },
+  };
+}
+
+export default function providerPluginExample(settings) {
+  // wrap your config with the ecommerce plugin
+  return new AcmeEcommercePlugin({
+    // ... same config as defined in the previous example
+  };)
+}
+```
+
+<div>
+
+:::info
+
+We handle this internally for
+[known plugins](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/src/web/core/analytics/plugins/e-commerce/e-commerce.js#L6-11)
+like the [`google-analytics`](https://getanalytics.io/plugins/google-analytics/)
+plugin.
+
+If you are aware of any other providers which require this,
+please&nbsp;<ContactLink>let us know</ContactLink> or feel free to contribute to
+our
+[analytics examples](https://gitlab.blackswift.cloud/front-commerce/examples/analytics).
+
+:::
+
+</div>
+
+## Using Plugins
```

Comment:
**suggestion:** maybe moving this section after the intro could help readers to better understand what `settings` are in the section that explains how to write your  own one.